### PR TITLE
[Spawner] Accept parsing multiple `--param-file` arguments to spawner 

### PR DESCRIPTION
**Pull request by saikishor**
When we use multiple controlers to activate or deactivate, it is quite cumbersome to accept only one `--param-file` arg as it is not very scalable. For this reason, now the spawner can accept multiple --param-file args and set the respective param_file to those corresponding controllers. 

Along with the above mentioned change, now the controllers can accept multiple param-file args as well, and this is very helpful to be able to override parameters between parameter files as how it is done with the standard ROS 2 nodes

### CODE REVIEW

**Comment by saikishor** [comment]
> Seems to be a good idea, and implementation LGTM.
> 
> Please update the [doc section](https://github.com/ros-controls/ros2_control/blob/master/controller_manager/doc/userdoc.rst#spawner) and release notes.

@christophfroehlich Done

File: controller_manager/doc/userdoc.rst [hunk 1]
```diff
@@ -157,9 +157,8 @@ There are two scripts to interact with controller manager from launch files:
 .. code-block:: console
 
     $ ros2 run controller_manager spawner -h
-    usage: spawner [-h] [-c CONTROLLER_MANAGER] [-p PARAM_FILE] [-n NAMESPACE] [--load-only] [--inactive] [-t CONTROLLER_TYPE] [-u]
-                      [--controller-manager-timeout CONTROLLER_MANAGER_TIMEOUT]
-                      controller_name
+    usage: spawner [-h] [-c CONTROLLER_MANAGER] [-p PARAM_FILE] [-n NAMESPACE] [--load-only] [--inactive] [-u] [--controller-manager-timeout CONTROLLER_MANAGER_TIMEOUT] [--activate-as-group]
```
Comment (christophfroehlich) [inline controller_manager/doc/userdoc.rst:160]: how can a user see from the help that -p can be given more than once? should we add this explicitely to the description string?

Comment (saikishor) [reply controller_manager/doc/userdoc.rst:160]: It is appended automatically internally. I can update the description of the parameter

Comment (saikishor) [reply controller_manager/doc/userdoc.rst:160]: Done

File: controller_manager/controller_manager/spawner.py [hunk 2]
```diff
@@ -79,8 +79,11 @@ def main(args=None):
     parser.add_argument(
         "-p",
         "--param-file",
-        help="Controller param file to be loaded into controller node before configure",
+        help="Controller param file to be loaded into controller node before configure. "
+        "Parse multiple times to load different files for different controllers or to "
```
Comment (christophfroehlich) [inline controller_manager/controller_manager/spawner.py:83]:
```suggestion
        "Pass multiple times to load different files for different controllers or to "
```
?

Comment (saikishor) [reply controller_manager/controller_manager/spawner.py]: Done

File: controller_manager/doc/userdoc.rst [hunk 3]
```diff
@@ -169,7 +168,7 @@ There are two scripts to interact with controller manager from launch files:
       -c CONTROLLER_MANAGER, --controller-manager CONTROLLER_MANAGER
                             Name of the controller manager ROS node
       -p PARAM_FILE, --param-file PARAM_FILE
-                            Controller param file to be loaded into controller node before configure
+                            Controller param file to be loaded into controller node before configure. Parse multiple times to load different files for different controllers or to override the parameters of the same controller.
```
Comment (saikishor) [inline controller_manager/doc/userdoc.rst:171]:
```suggestion
                            Controller param file to be loaded into controller node before configure. Pass multiple times to load different files for different controllers or to override the parameters of the same controller.
```

**Review by christophfroehlich — APPROVED** [review]
Thx :)

**Review by christophfroehlich — APPROVED** [review]

**Review by christophfroehlich — APPROVED** [review]

File: hardware_interface/include/hardware_interface/controller_info.hpp [hunk 4]
```diff
@@ -34,7 +34,7 @@ struct ControllerInfo
   std::string type;
 
   /// Controller param file
-  std::optional<std::string> parameters_file;
+  std::optional<std::vector<std::string>> parameters_files;
```
[the remark this answers is not in the document]
Comment (bmagyar) [reply hardware_interface/include/hardware_interface/controller_info.hpp:37]: created a good first issue https://github.com/ros-controls/ros2_control/issues/1893

[the remark this answers is not in the document]
Comment (saikishor) [reply hardware_interface/include/hardware_interface/controller_info.hpp:37]: True. You are right. Can be done that way😅

**Review by bmagyar — APPROVED** [review]

**Pull request merged**